### PR TITLE
Add initial fuzzing for PE/COFF parsing in libunwindstack

### PR DESCRIPTION
**Pull request by ronaldfw**
This adds a first fuzzer for PE/COFF parsing. The fuzzer is extremely
basic and generates very poor coverage as it is not at all aware of
the structure of PE/COFF files. This will be fixed later. For now,
this is added to establish the structure of how we add fuzzing tests
for libunwindstack and to make sure it shows up in our oss-fuzz
integration. Next steps will be to add a custom mutator and/or add a
corpus of PE/COFF files to trigger more code paths.

Tested: Ran fuzzer locally.
Bug: http://b/192514457

### CODE REVIEW

File: third_party/libunwindstack/tests/fuzz/PeCoffInterfaceFuzzer.cpp [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+#include <unwindstack/Memory.h>
+#include <unwindstack/PeCoffInterface.h>
+
+// The most basic fuzzer for PE/COFF parsing. Generates really poor coverage as
+// it is not PE/COFF file structure aware.
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  std::shared_ptr<unwindstack::Memory> memory =
+      unwindstack::Memory::CreateOfflineMemory(data, 0, size);
+  unwindstack::PeCoffInterface<uint64_t> pe_coff_interface(memory.get());
```
Comment (ronaldfw) [inline third_party/libunwindstack/tests/fuzz/PeCoffInterfaceFuzzer.cpp:29]: Call to `Init()` is missing here. I'll fix.

**Review by beckerhe — APPROVED** [review]
Thank you. LGTM.

On a more general note: I tried this out on my machine and the fuzzer produces a lot of logging to stdout. That severely slows down the fuzzing since basically every iteration results in an error message printed to stdout. I'm not sure how to solve this since you're only following the style of the other code in the library which is doing the same. Something like ErrorMessageOr doesn't exist. Not sure what's the best solution here. Of course we could remove the logging. Or we could tweak LogStdout.cpp such that we can disable printing to stdout via a global atomic bool. (More something to think about rather than a request to change anything in this PR.)

**Comment by ronaldfw** [comment]
Thanks for the review!

For the output spam, libfuzzer actually has a solution: One can pass the option -close_fd_mask=3 to mute both stdout and stderr (or 0, 1, 2 approriately to mask whatever is desired). We can also set this in our oss-fuzz project, for example by providing a default.options file like here: https://github.com/google/oss-fuzz/blob/master/projects/radare2/default.options

**Comment by beckerhe** [comment]
> Thanks for the review!
> 
> For the output spam, libfuzzer actually has a solution: One can pass the option -close_fd_mask=3 to mute both stdout and stderr (or 0, 1, 2 approriately to mask whatever is desired). We can also set this in our oss-fuzz project, for example by providing a default.options file like here: https://github.com/google/oss-fuzz/blob/master/projects/radare2/default.options

Oh wow. That's great. I haven't heard about this. Sounds like a good solution!

**Comment by ronaldfw** [comment]
> > Thanks for the review!
> > For the output spam, libfuzzer actually has a solution: One can pass the option -close_fd_mask=3 to mute both stdout and stderr (or 0, 1, 2 approriately to mask whatever is desired). We can also set this in our oss-fuzz project, for example by providing a default.options file like here: https://github.com/google/oss-fuzz/blob/master/projects/radare2/default.options
> 
> Oh wow. That's great. I haven't heard about this. Sounds like a good solution!

Huh, maybe I was too fast, from the documentation: "Be careful, this will also close e.g. stderr of asan." I guess this means that we lose some useful output. Not sure to what degree this matters for our oss-fuzz integration.

**Comment by beckerhe** [comment]
> > > Thanks for the review!
> > > For the output spam, libfuzzer actually has a solution: One can pass the option -close_fd_mask=3 to mute both stdout and stderr (or 0, 1, 2 approriately to mask whatever is desired). We can also set this in our oss-fuzz project, for example by providing a default.options file like here: https://github.com/google/oss-fuzz/blob/master/projects/radare2/default.options
> > 
> > 
> > Oh wow. That's great. I haven't heard about this. Sounds like a good solution!
> 
> Huh, maybe I was too fast, from the documentation: "Be careful, this will also close e.g. stderr of asan." I guess this means that we lose some useful output. Not sure to what degree this matters for our oss-fuzz integration.

Closing stdout should be enough to silence `Log::Error` from libunwindstack. So this should not affect the asan and libfuzzer output since both go to stderr (as you said).

**Comment by ronaldfw** [comment]
Yes, you are right, that's great. I'll prepare a change for oss-fuzz.

**Pull request merged**